src/types.cpp

```diff
@@ -256,7 +256,7 @@ SizedType CreateInteger(size_t bits, bool is_signed)
   // enough information to figure out the exact size of the integer. Later
   // passes infer the exact size.
   assert(bits == 0 || bits == 1 || bits == 8 || bits == 16 || bits == 32 ||
-         bits == 64);
+         bits == 64 || bits == 128);
```

Comment:
Do you know what happens in codegen if something tries to use a 128-bit integer?

Reply:
Let's add a test and figure it out...

Reply:
@ajor Following up on our chat, I modified `SizedType` to accept 128-bits integers. However, they get truncated down to only 64-bits with a warning to the user. The warning is done by the `semantic_analyser`, then the `codegen_llvm` insert the truncate for a `FieldAccess` on a user-defined type, if required.